Dashboard: Welcome panel refresh (jazzy version)

ryelle opened the pull request:
This is the "jazzy" design for the new Welcome panel, with new content. This does not include the textured background, I don't know how to export it from figma. I'll need a repeating texture, as small a filesize as possible :)

I had to do some fussy positioning to get the balloons & columns to look good/be readable across screen sizes. Here are a few breakpoint screenshots:

![dash-1440](https://user-images.githubusercontent.com/541093/145870098-8ddc26f4-a2fc-469b-8a9c-435ab2929a74.png)

![dash-1080](https://user-images.githubusercontent.com/541093/145870096-ae5e91e7-68d3-4118-a9b0-dbe7b2d3462e.png)

![dash-782](https://user-images.githubusercontent.com/541093/145870095-1f5ad2d5-fd69-45a9-b345-7c669a7b71a9.png)

![dash-480](https://user-images.githubusercontent.com/541093/145870093-0e8109fe-f827-42d1-b422-8de624458f49.png)

Unfortunately there are still some intermediate sizes where the balloons & text overlap:

![dash-1200-overlap](https://user-images.githubusercontent.com/541093/145870446-24966e99-0986-452a-a47c-be945f3e3918.png)

Trac ticket: https://core.trac.wordpress.org/ticket/54489

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

ryelle committed:
Dashboard: Welcome panel refresh (jazzy version)

ryelle committed:
Fix linter issues

critterverse commented:
Thanks so much for working on this, @ryelle! You're right that the layout isn't working great across every size 😅  I took a pass on these breakpoint sizes in Figma to see if we could get the balloons and text to play a little more nicely together: https://www.figma.com/file/MfN5ccveFY5RncaUaegNec/Welcome-banner?node-id=119%3A2444

1440:

![1440](https://user-images.githubusercontent.com/77359364/146108273-f0d495e8-e1cc-484c-9255-ce4135368c9f.png)

1080:

![1080](https://user-images.githubusercontent.com/77359364/146108279-d9b80880-9faa-416f-b964-e43bc9f4606e.png)

782:

![782](https://user-images.githubusercontent.com/77359364/146117070-9979abe0-eb40-46e7-b021-e421e1b32eaf.png)

480:

![480](https://user-images.githubusercontent.com/77359364/146108307-6c31bb39-5bc8-4c31-b8a4-4dfd010c1ff8.png)

A few notes:

- I wonder if we should create a top area to contain the balloon artwork and texture within. This area could remain the same height across breakpoints.
- When responding down, ideally the balloon image could shrink about 10% between each breakpoint and stay anchored to the upper right (the size/position of the balloons has also been updated from the previous version). 
- The icons don't seem to work as well when stacked vertically with the text so it would be nice if they could stay locked to the left of each blurb. Would it be weird to hide them between the two largest breakpoints when the spacing gets too tight? They could show again on the smaller sizes once the horizontal columns stack into a single column. 
- When the layout switches to a single column around 782 px, we could limit the text from going the full width of the banner to help with line length/readability. 
- Can we maintain a minimum distance between the top text and bottom text so there’s always some breathing room between the two sections? Noticed that it was getting tight on the smaller screen sizes and think the banner could be a bit longer in those contexts.

Trac ticket: https://core.trac.wordpress.org/ticket/54489

ryelle committed:
Update dismiss hover/focus color

ryelle committed:
Update to match new design, add texture

ryelle committed:
Update the line-length on tablet-sized screens

ryelle commented:
I think I've addressed the feedback - I aimed to match the mockups. I also figured out how to export the texture and made a smaller repeating version 🎉 

The new layout adds a bit more space between the heading and the columns, but I'm not sure if that matches what you're describing, @critterverse, so let me know if this needs more work.

1440
![update-1440](https://user-images.githubusercontent.com/541093/146248755-b8123207-f806-4630-867b-c104f71f73ca.png)

1080
![update-1080](https://user-images.githubusercontent.com/541093/146248753-33fc27fb-726d-4d8b-bf7d-06f325da8ed4.png)

782
![update-782](https://user-images.githubusercontent.com/541093/146248750-21901258-f10f-457d-bf45-d889e350a753.png)

480
![update-480](https://user-images.githubusercontent.com/541093/146248746-c2830e9e-eca8-4237-bccf-dc0edad789c7.png)

ryelle commented:
I introduced a bug on very large screens… will circle back to clean this up later, along with any other design feedback.

<img width="364" alt="Screen Shot 2021-12-15 at 2 11 11 PM" src="https://user-images.githubusercontent.com/541093/146249911-bf8d4b31-0b06-490c-b23f-114f071f7402.png">

ryelle commented:
Also for reference, layout looks okay in RTL language:

<img width="1322" alt="Screen Shot 2021-12-15 at 2 13 38 PM" src="https://user-images.githubusercontent.com/541093/146250258-7bd10959-e8bd-4d5c-8844-37a828e7b9cd.png">

critterverse commented:
Wow this looks so good, thank you @ryelle! 💙

I only had a couple of notes —

- Could we still keep the icons to the left of the text (rather than vertically stacked) in the 480 view?
- It's nice how the word "Dismiss" next to the close button is hidden at the smallest breakpoint — could this happen sooner around 782px? I'm thinking that would help with some of the intermediate sizes around the 650 range if the balloons could be a bit closer to the right edge of the banner:
 
<img width="648" alt="banner_650" src="https://user-images.githubusercontent.com/77359364/146419878-24f9af81-8d48-4824-a71d-a670dade5661.png">

^ Side note that I wasn’t able to see the texture when testing but I’m sure this is probably something I did wrong with the setup — it looks great in your screenshots above!

ryelle committed:
Remove leftover style, doesn't apply to anything visible

ryelle committed:
Hide dismiss button at higher breakpoint, move balloons closer to the edge

ryelle committed:
Leave icons next to text at mobile breakpoint, hide icons at very small screen size

ryelle commented:
Okay, I've change the breakpoint where "dismiss" disappears, so we can move the balloons over a little early. I also put the icons back next to the text on the mobile breakpoint, but also added another "very small" breakpoint where the icons are hidden & the text is full-width.

![Screen Shot 2021-12-16 at 15 13 25](https://user-images.githubusercontent.com/541093/146442677-1b5792ae-6f05-4b2f-a6d4-3fdbd620c357.png)
![Screen Shot 2021-12-16 at 15 13 30](https://user-images.githubusercontent.com/541093/146442680-d09bddec-0453-4e7d-8dfe-026ab7e47d79.png)
![Screen Shot 2021-12-16 at 15 15 38](https://user-images.githubusercontent.com/541093/146442930-f4ddb43c-e2f3-4006-8570-7d09eec4e10d.png)

I'm not sure why the texture wouldn't show up for you, but it might be how you're applying the patch? You can [download it directly](https://github.com/WordPress/wordpress-develop/blob/b7d1aa292af07f0d5b35f0854df2b15ff6af04d6/src/wp-admin/images/about-texture.png) and drop it in the folder `src/wp-admin/images` (or `build/wp-admin/images`).

critterverse commented:
Awesome, this looks great 🎉  Here's what I'm seeing:

https://user-images.githubusercontent.com/77359364/146602123-a42b4591-3a58-4949-b99c-dfa5c2fe5977.mov

I'll drop this in the Design channel on the Making Wordpress Slack as well for more visibiity!

melchoyce commented:
How's this work on other admin color schemes? Any way to update the background and text color depending on which color scheme's activated? 👀

ryelle commented:
Screenshots from a few other schemes.

![Screen Shot 2021-12-20 at 12 30 57](https://user-images.githubusercontent.com/541093/146808256-e457b881-8e10-4a47-9ba7-8099dab32542.png)
![Screen Shot 2021-12-20 at 12 31 11](https://user-images.githubusercontent.com/541093/146808257-f9821a03-a6de-4323-a2fb-6b22624bf466.png)
![Screen Shot 2021-12-20 at 12 31 25](https://user-images.githubusercontent.com/541093/146808259-4a948b98-6ae9-492c-837a-fbacb8ded12e.png)
![Screen Shot 2021-12-20 at 12 31 33](https://user-images.githubusercontent.com/541093/146808260-78324b4a-366b-461e-becd-02736c7c64a9.png)

> Any way to update the background and text color depending on which color scheme's activated?

That sounds fun, but this is meant to mirror the about page and I don't think that's in the plans there. On the other hand, if the about page style will stay consistent over releases, maybe it would be good to bring in the color scheme colors. @critterverse any thoughts about this?

ryelle committed:
Fix columns background on very large screens

ryelle committed:
Use the SVG from the about page

ryelle committed:
Update copy for non-block-theme themes

ryelle committed:
Put the full link into the translation function, so it can be updated to rosetta sites

ryelle commented:
Latest iteration, if a block theme is active:
<img width="1137" alt="block-theme" src="https://user-images.githubusercontent.com/541093/148083862-c9151346-a7dd-4f5e-bae1-55380fca4b8a.png">

If a classic theme is active:
<img width="1138" alt="non-block-theme" src="https://user-images.githubusercontent.com/541093/148083855-b19c1700-dc97-4c89-bba6-1022ad62f057.png">

Note: that last link goes to a draft support article, a [user support doc for block themes](https://github.com/WordPress/Documentation-Issue-Tracker/issues/63), and I'm just assuming the URL. I'm pretty sure it should be in the string for translation so it can be updated to the locale site if available, but I can remove it if not.

hellofromtonya commented on src/wp-admin/includes/dashboard.php:
```diff
@@ -1990,70 +1990,61 @@ function wp_dashboard_empty() {}
+			<?php else : ?>
+				<h3><?php echo ( 'Discover a new way to build your site.' ); ?></h3>
+				<p><?php echo ( 'There&#8217;s a new kind of WordPress theme, called a block theme, that lets you build the site you&#8217;ve always wanted &#8212; with blocks and styles.' ); ?></p>
+				<?php echo ( '<a href="https://wordpress.org/support/article/block-themes/">Learn about block themes</a>' ); ?>
```
>Note: that last link goes to a draft support article, a user support doc for block themes, and I'm just assuming the URL. I'm pretty sure it should be in the string for translation so it can be updated to the locale site if available, but I can remove it if not.

Yes, the link should be translatable. But like you said, it's a draft support article. What's been done in the past when the article doesn't yet exist? Is it added later? Or ____?

hellofromtonya replied:
[Confirmed with @annezazu that the link is correct](https://github.com/WordPress/Documentation-Issue-Tracker/issues/63#issuecomment-1005008936). 🎉

critterverse commented:
Thanks @ryelle, this is looking great!

Definitely an edge case but on super large screen sizes — could we make the blue background (without the noise texture) extend edge-to-edge?

![XL-viewport](https://user-images.githubusercontent.com/77359364/148088309-ce9bf48b-7f2a-495c-a661-472f4702b869.png)

>Any way to update the background and text color depending on which color scheme's activated?

This is a really fun idea @melchoyce @ryelle 😁 For the sake of trying it out, the color coordination system could be:

- The background color behind the 3 columns of text would always be white
- The background color of the top textured area comes from the highlight color of the active menu item in wp-admin
- The text color in the top textured area comes from the text color of the active menu text in wp-admin
- The background color for the icons comes from the color of the wp-admin sidebar
- The icon color comes from the text color of the default/inactive menu text in wp-admin

![schemes-1](https://user-images.githubusercontent.com/77359364/148085877-3b4744a8-c639-4ca4-b00c-15779c11cdb6.png)

Another possibility is the reverse (which may have slightly better contrast but it's hard to say universally):

- The background color behind the 3 columns of text would always be white
- The background color of the top textured area comes from the color of the wp-admin sidebar
- The text color in the top textured area comes from the color of the default/inactive menu text in wp-admin
- The background color for the icons comes from the highlight color of the active menu item in wp-admin
- The icon color comes from the color of the active menu text in wp-admin

![schemes-2](https://user-images.githubusercontent.com/77359364/148086052-d4531bde-e236-40be-83d2-5becbdf2679c.png)

The colors do feel more harmonious and the design still ties in nicely with the About page through the graphic elements/texture. The main drawbacks IMO are that some of the color schemes don't pass color contrast standards and a few feel a bit dated. 

If we do stick with the Blueberry color, it might help to use the white background behind the columns of text as shown in the examples above so there's a smaller area of color.

What do others think?

ryelle commented:
> Definitely an edge case but on super large screen sizes — could we make the blue background (without the noise texture) extend edge-to-edge?

Are you on the latest version of the PR? that was fixed already. Maybe try clearing cache and refreshing?

For today let's focus on getting this PR as it is committed, since we're running against string freeze in 3hrs. I can try playing around with the color schemes later this week if there's interest (I do like it 😁 )

> The main drawbacks IMO are that some of the color schemes don't pass color contrast standards and a few feel a bit dated.

hah, fair point. About contrast, that's fine since these are opt-in themes, not the default (also, some people prefer lower contrast themes, that's why we have options).

melchoyce commented:
Love that first set! Using the highlight colors really brings attention to that section 😃

costdev commented on src/wp-admin/includes/dashboard.php:
```diff
@@ -1990,70 +1990,61 @@ function wp_dashboard_empty() {}
+			<div class="welcome-panel-icon-pages"></div>
+			<div class="welcome-panel-column-content">
+				<h3><?php _e( 'Author rich content with blocks and patterns' ); ?></h3>
+				<p><?php _e( 'Block patterns are pre-configured block layouts. Use them to get inspired or create new pages in a flash.' ); ?></p>
```
```suggestion
				<p><?php _e( 'Block patterns are pre&#8209;configured block layouts. Use them to get inspired or create new pages in a flash.' ); ?></p>
```

Just an initial suggestion to convert all instances of `-` to `&#8209;` for the strings.
Similarly, `’` should be converted to `&#8217;`.

ryelle replied:
I think I got all the `’`s. I missed 2 emdashes, so those have been updated. I think this instance should stay a normal hyphen though, not a non-breaking hyphen. Does that still need to be encoded?

ryelle committed:
Use html entities for emdashes

hellofromtonya commented on src/wp-admin/includes/dashboard.php:
```diff
@@ -1990,70 +1990,61 @@ function wp_dashboard_empty() {}
+				<p><?php _e( 'Tweak your site, or give it a whole new look! Get creative — how about a new color palette or font?' ); ?></p>
+				<a href="<?php echo esc_url( admin_url( 'site-editor.php?styles=open' ) ); ?>"><?php _e( 'Edit styles' ); ?></a>
+			<?php else : ?>
+				<h3><?php echo ( 'Discover a new way to build your site.' ); ?></h3>
+				<p><?php echo ( 'There&#8217;s a new kind of WordPress theme, called a block theme, that lets you build the site you&#8217;ve always wanted &#8212; with blocks and styles.' ); ?></p>
+				<?php echo ( '<a href="https://wordpress.org/support/article/block-themes/">Learn about block themes</a>' ); ?>
```
```suggestion
				<h3><?php _e( 'Discover a new way to build your site.' ); ?></h3>
				<p><?php _e( 'There&#8217;s a new kind of WordPress theme, called a block theme, that lets you build the site you&#8217;ve always wanted &#8212; with blocks and styles.' ); ?></p>
				<a href="<?php _e( 'https://wordpress.org/support/article/block-themes/' ); ?>"><?php _e( 'Learn about block themes' ); ?></a>
```

audrasjb replied:
@hellofromtonya  line 2044 : rather use `echo esc_url( __( 'https://wordpress.org/support/article/block-themes/' ) )`

audrasjb commented:
Just noting that translatable url in html links should rather use this kind of code:
`echo esc_url( __( 'https://wordpress.org/support/article/block-themes/' ) )`
(see example on line 2044)

ryelle committed:
Replace the placeholder functions with real translation functions

hellofromtonya reviewed (approved):
LGTM! @ryelle ready for commit 👍

critterverse commented:
> Are you on the latest version of the PR? that was fixed already. Maybe try clearing cache and refreshing?

My mistake, this looks good ✅

> For today let's focus on getting this PR as it is committed, since we're running against string freeze in 3hrs. I can try playing around with the color schemes later this week if there's interest (I do like it 😁 )

Sounds like a plan! Leaving one more possibility for the color scheme-specific version here to make sure I've exhausted all the possibilites 😂 This version uses the color more sparingly so the the balloon image asset could always be the Blueberry color (i.e. a hybrid approach) or it could use the scheme highlight color if possible:

![schemes-3A](https://user-images.githubusercontent.com/77359364/148118785-70f5eb78-3040-4cda-b20c-c692ef03ccf7.png)

JJJ commented:
A wandering thought... will there be (is there already) default styling for non-core color schemes?

<img width="1351" alt="jjj-on-2022-01-05-at-15-36-48@2x" src="https://user-images.githubusercontent.com/88951/148293642-1f0bf21c-eeb4-46ca-9850-2d666c6a6fea.png">
